internal/services/automation/automation_software_update_configuration_resource_test.go

```diff
@@ -451,7 +623,7 @@ provider "azurerm" {
 
 resource "azurerm_resource_group" "test" {
   name     = "acctestRG-auto-%[1]d"
-  location = "West US"
```

Comment:
Use a random location will cause test failure, because the [required log analytics workspace](https://aka.ms/Automation-Region-Mapping) location is not the same as the automation account's location in `EastUS` and  `EastUS2`.

![image](https://github.com/hashicorp/terraform-provider-azurerm/assets/2633022/b99e0563-ab10-4c61-b615-bd309e36f64a)

Error as below;
```
Original Error: autorest/azure: Service returned an error. Status=400 Code="BadRequest" Message="Error - chosen Azure Automation does not have a Log Analytics workspace linked for operation to succeed. To know which regions are supported for linking, see our documentation at: https://aka.ms/Automation-Region-Mapping
```